src/test/unit/mcmc/hmc/base_hmc_test.cpp

```diff
@@ -124,7 +124,7 @@ TEST(McmcBaseHMC, streams) {
   stan::interface_callbacks::writer::stream_writer writer(output);
 
   EXPECT_NO_THROW(sampler.write_sampler_state(writer));
-  EXPECT_EQ("Step size = 0.1\nNo free parameters for unit metric\n",
+  EXPECT_EQ("Step size = 0.1\n",
```

Comment:
(fix now) We don't want to hard code line returns.  If we use `std::endl` then the newline will be platform specific.  So I'd just test without the newline `\n` symbol.  

(future) Avoid newlines in exception messages.